agent/worker.go

```diff
@@ -500,8 +505,17 @@ func (w *worker) newTaskManager(ctx context.Context, tx *bolt.Tx, task *api.Task
 // updateTaskStatus reports statuses to listeners, read lock must be held.
 func (w *worker) updateTaskStatus(ctx context.Context, tx *bolt.Tx, taskID string, status *api.TaskStatus) error {
 	if err := PutTaskStatus(tx, taskID, status); err != nil {
+		// we shouldn't fail to put a task status. however, there exists the
+		// possibility of a race in which we try to put a task status after the
+		// task has been deleted. because this whole contraption is a careful
+		// dance of too-tightly-coupled concurrent parts, fixing tht race is
+		// fraught with hazards. instead, we'll recognize that it can occur,
+		// log the error, and then ignore it.
 		log.G(ctx).WithError(err).Error("failed writing status to disk")
```

Comment:
Ah, sorry, just realised this: so we're always logging this as an error, and in case its `!errTaskUnknown`, we're _both_ logging it here, _and_ returning the error.

I guess it could be something like this (and log as Debug, Info, or Warn?);

```go
if err == errTaskUnknown {
	// we shouldn't fail to put a task status. however, there exists the
	// possibility of a race in which we try to put a task status after the
	// task has been deleted. because this whole contraption is a careful
	// dance of too-tightly-coupled concurrent parts, fixing tht race is
	// fraught with hazards. instead, we'll recognize that it can occur,
	// log the error, and then ignore it.
	log.G(ctx).WithError(err).Debug("failed writing status to disk")
	return nil
}
return err
```

Reply:
Yeah, you're right. I'm gonna log it at Info level, and I added a comment explaining why.

Reply:
Thx!